### DISCUSSION TITLE
Company suggestion popup disappears when only one candidate remains

### DISCUSSION THREAD

**Discussion by cmatzenbach** [company-mode/company-mode#1194] [Q&A] [answered]
![company-popup-disappear](https://user-images.githubusercontent.com/15268289/46621251-9c68b300-caec-11e8-8f6d-0bf7334e7e8b.gif)

So this has been bothering me for awhile - anytime I have one completion remaining, company drops the menu (but will complete the term if you run company-complete, which I do in the end of this .gif).  I've searched around but can't find anything on it.  I disabled all the packages that could affect company before recording the gif (as you see, I typed 'asse' until 'assert' was the only choice remaining, and the completion window disappeared - pressing my binding for company-complete is what completed it).  However, I find this functionality very annoying, because I never know if I misspelled or if I am correct and there is a candidate ready to be completed.  Is there some setting I can set to change this?  Please let me know if you need any additional info.

**Comment by noctuid** [comment]
You need to customize `company-frontends` (e.g. See [here](https://emacs.stackexchange.com/questions/36619/show-tooltip-even-if-there-is-only-one-candidate-in-company-mode)).

**Comment by dgutov** [comment, marked as answer]
@cmatzenbach It looks like there's an element missing from `company-frontends` (which is there in the default set). It's used to display the remaining sole completion inline. `company-preview-if-just-one-frontend`.

**Comment by cmatzenbach** [comment]
@dgutov Thanks, that worked.
@noctuid What I _really_ wanted was exactly what the poster in the link you posted wanted, but setting company-frontends as described by the accepted answer does not show the pop-up for one remaining word - in fact, it shows nothing at all.  Did you get that to work (the pop-up for one candidate), or did you just have the link handy?  If not, no worries, I can live with @dgutov 's answer, but yeah, that's exactly what I wanted.  I'm surprised the poster said that worked - those two variables do nothing to preview when one candidate remains.

**Comment by dgutov** [comment]
> company-frontends as described by the accepted answer does not show the pop-up for one remaining word

That's odd. The one that hides the popup in this case is `company-pseudo-tooltip-unless-just-one-frontend`, and not `company-pseudo-tooltip-frontend` mentioned there.
